### ISSUE TITLE
Allow scaling icons in theme editor

### ISSUE THREAD

**Issue by nonunknown** [godotengine/godot-proposals#5414]
### Describe the project you are working on

Godot Kart Racing

### Describe the problem or limitation you are having in your project

I'm creating a custom theme, now for checkbox button, and I'm not allowed to scale the checkbox icon!

![image](https://user-images.githubusercontent.com/58845030/189885721-5a881977-4efa-429d-bede-30f18a2874bc.png)

implementing this should fix https://github.com/godotengine/godot-proposals/issues/4442

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a property to scale the icons for any node that uses this feature

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

a slider or spinbutton to scale icons

### If this enhancement will not be used often, can it be worked around with a few lines of script?

no, as far as I know

### Is there a reason why this should be core and not an add-on in the asset library?

core feature

**Comment by YuriSizov** [comment]
What do you mean by scaling them? Do you want to change the texture itself somehow? Or do you want to change the preview?

**Comment by YuriSizov** [comment]
Yes, but the other linked proposal, https://github.com/godotengine/godot-proposals/issues/4442, implies that they want some way to control the scale precisely. Which is not related to the Theme editor. Preview you can indeed customize to your needs, but it cannot be do something that controls don't already support.

**Comment by nonunknown** [comment]
@YuriSizov I want to modify the checkbox size to any size I want, take button as an example, you can modify everything on it, Azagaya had an awesome idea, which is turning the button icons into styleboxes, that way we can customize anyway we want!

**Comment by YuriSizov** [comment]
This is completely unrelated to the Theme editor. The Theme editor lets you edit the `Theme` resource. It supports what controls support, and gives you tools to edit it. If you want more control over the icon of checkbox/checkbutton, then that's what you need to make a proposal about.

**Comment by QueenOfSquiggles** [comment]
I like this proposal in general. I'll await the proposal compliant with @YuriSizov 's specifications <3

**Comment by nonunknown** [comment]
@YuriSizov sorry but I have no idea how to be more specific than that! why you can customize button in any way, but you cant do for its checkbox? gotcha!!!

**Comment by YuriSizov** [comment]
@nonunknown You can customize CheckBox in the same ways you can customize Button, because CheckBoxes are Buttons and they have all the same theme properties.

**Comment by nonunknown** [comment]
I'm talking about the checkbox/radio texture itself, that's what I want to resize

**Comment by nonunknown** [comment]
> Downscaling textures significantly should be avoided for two reasons:
> 
> * It results in aliasing if mipmaps are disabled (which is the default in 2D in Godot, as it saves memory when mipmaps are not needed).
> * It wastes memory by never displaying a texture at its full size.

actually if you want a proper size for your checkbox/radio texture you have to do that, and this proposal solves that

**Comment by Calinou** [comment]
Note that a workaround using MeshTexture exists: https://github.com/godotengine/godot-proposals/discussions/9080#discussioncomment-10988411
